helix-term/src/application.rs

```diff
@@ -663,29 +673,32 @@ impl Application {
                         }))
                     }
                     MethodCall::WorkspaceFolders => {
-                        let language_server =
-                            self.editor.language_servers.get_by_id(server_id).unwrap();
-
-                        Ok(json!(language_server.workspace_folders()))
+                        Ok(json!(get_language_server!().workspace_folders()))
                     }
                     MethodCall::WorkspaceConfiguration(params) => {
+                        let language_server = get_language_server!();
                         let result: Vec<_> = params
                             .items
                             .iter()
                             .map(|item| {
-                                let mut config = match &item.scope_uri {
-                                    Some(scope) => {
-                                        let path = scope.to_file_path().ok()?;
-                                        let doc = self.editor.document_by_path(path)?;
-                                        doc.language_config()?.config.as_ref()?
-                                    }
-                                    None => self
-                                        .editor
-                                        .language_servers
-                                        .get_by_id(server_id)
-                                        .unwrap()
-                                        .config()?,
-                                };
+                                let mut config = language_server.config()?;
+                                // TODO check for scope_uri (below) necessary?
+                                // let mut config = match &item.scope_uri {
+                                //     Some(scope) => {
+                                //         let path = scope.to_file_path().ok()?;
+                                //         let doc = self.editor.document_by_path(path)?;
+                                //         doc.language_config()?
+                                //             .language_servers
+                                //             .as_ref()
+                                //             .and_then(|ls| ls.config.as_ref())?
+                                //     }
+                                //     None => self
+                                //         .editor
+                                //         .language_servers
+                                //         .get_by_id(server_id)
+                                //         .unwrap()
+                                //         .config()?,
+                                // };
```

Comment:
I'm not sure if this (`scope_uri`) is necessary at all.

I don't think a language server requires any other configuration than the configuration for itself?

Reply:
I think I [initially did this](https://github.com/helix-editor/helix/pull/1684) in order to be [spec compliant](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration) and return nothing in case the document is not opened in Helix. I can't think of any situation where this would actually be relevant though, so it seems best to ignore the scope (until we support for example EditorConfig). Maybe remove it instead of just commenting it out?

Reply:
I removed the comment in the latest rebase, thanks for letting me know.